contracts/cw1-subkeys/src/contract.rs

```diff
@@ -82,14 +83,38 @@ pub fn handle_increase_allowance<S: Storage, A: Api, Q: Querier, T>(
 where
     T: Clone + fmt::Debug + PartialEq + JsonSchema,
 {
-    // placeholder to remove warnings
-    let _ = (deps, env, spender, amount, expires);
-
-    // TODO
-    // look at cw20_base::contract::IncreaseAllowance
-    // if nothing present, we set this, and expires=None => Expiration::Never
-    // if something present, we add to the balance. expires=None => leave expiration at previous state, expires=Some(x) => set expires to new value x
-    panic!("unimplemented")
+    let cfg = config_read(&deps.storage).load()?;
+    let spender_raw = &deps.api.canonical_address(&spender)?;
+    let owner_raw = &deps.api.canonical_address(&env.message.sender)?;
+
+    if !cfg.is_admin(&owner_raw) {
+        return Err(StdError::unauthorized());
+    }
+    if spender_raw == owner_raw {
+        return Err(StdError::generic_err("Cannot set allowance to own account"));
+    }
+
```

Comment:
Good logic here

Reply:
Thanks. Copied it from erc20-base, I think.